src/org/zaproxy/zap/extension/exportreport/ExportReportAPI.java

```diff
@@ -0,0 +1,184 @@
+
+            if (alertDetailsFlags.size() != extension.maxList) {
+                logger.error(Constant.messages.getString("exportreport.message.console.error.details", Constant.messages.getString("exportreport.menu.details.label"), alertDetailsFlags.size(), extension.maxList, Constant.messages.getString("exportreport.details.cweid.label"), Constant.messages.getString("exportreport.details.wascid.label"), Constant.messages.getString("exportreport.details.description.label"), Constant.messages.getString("exportreport.details.otherinfo.label"), Constant.messages.getString("exportreport.details.solution.label"), Constant.messages.getString("exportreport.details.reference.label"), Constant.messages.getString("exportreport.details.requestheader.label"), Constant.messages.getString("exportreport.details.responseheader.label"), Constant.messages.getString("exportreport.details.requestbody.label"), Constant.messages.getString("exportreport.details.responsebody.label")));
+                return ApiResponseElement.FAIL;
+            }
+            if (logger.isDebugEnabled()) logger.debug(Constant.messages.getString("exportreport.message.console.info.length", Constant.messages.getString("exportreport.menu.details.label"), Constant.messages.getString("exportreport.message.console.info.status.valid")));
+
+            if (!extension.validList(alertDetailsFlags)) {
+                logger.error(Constant.messages.getString("exportreport.message.console.error.valid.list", Constant.messages.getString("exportreport.menu.details.label")));
+                return ApiResponseElement.FAIL;
+            }
+            if (logger.isDebugEnabled()) logger.debug(Constant.messages.getString("exportreport.message.console.info.content", Constant.messages.getString("exportreport.menu.details.label"), Constant.messages.getString("exportreport.message.console.info.status.valid")));
+
+            if (logger.isDebugEnabled()) logger.debug(Constant.messages.getString("exportreport.message.console.info.pass.generate"));
+
+            ArrayList<String> alertSeverityTemp = extension.generateList(alertSeverityFlags, extension.getAlertSeverity());
+
+            ArrayList<String> alertDetailsFull = new ArrayList<String>();
+            alertDetailsFull.addAll(0, extension.getAlertDetails());
+            alertDetailsFull.addAll(extension.getAlertDetails().size(), extension.getAlertAdditional());
+            ArrayList<String> alertDetailsTemp = extension.generateList(alertDetailsFlags, alertDetailsFull);
+
+            try {
+                if (extension.generateReport(absolutePath, fileExtension, sourceDetails, alertSeverityTemp, alertDetailsTemp)){
+                    if (logger.isDebugEnabled()) logger.debug(Constant.messages.getString("exportreport.message.console.info.pass.path", absolutePath));
+                    return ApiResponseElement.OK;
+                }
```

Comment:
It should return something in case `generateReport(...)` returns false. Otherwise it leads to `BAD_ACTION` which is not the expected.